x/stakeibc/keeper/msg_server_rebalance_validators.go

```diff
@@ -61,35 +39,34 @@ func (k msgServer) RebalanceValidators(goCtx context.Context, msg *types.MsgReba
 
 	// we convert the above map into a list of tuples
 	type valPair struct {
-		deltaAmt int64
+		deltaAmt sdk.Int
 		valAddr  string
 	}
 	valDeltaList := make([]valPair, 0)
-	for _, valAddr := range utils.StringToIntMapKeys(validatorDeltas) {
-		deltaAmt := validatorDeltas[valAddr]
-		k.Logger(ctx).Info(fmt.Sprintf("Adding deltaAmt: %d to validator: %s", deltaAmt, valAddr))
+	for valAddr, deltaAmt := range validatorDeltas {
```

Comment:
Why are you able to remove the deterministic loop here? Is it because the sort is now now deterministic with .LT ?

Reply:
Great catch, we should not be removing this! We should really be using the informal systems gosec to catch these issues.

Reply:
[.../stride/x/stakeibc/keeper/msg_server_rebalance_validators.go:46-49] - G705 (CWE-): expected exactly 1 statement (either append, delete, or copying to another map) in a range with a map, got 2 (Confidence: MEDIUM, Severity: HIGH)
    45:         valDeltaList := make([]valPair, 0)
  > 46:         for valAddr, deltaAmt := range validatorDeltas {
  > 47:                 k.Logger(ctx).Info(fmt.Sprintf("Adding deltaAmt: %v to validator: %s", deltaAmt, valAddr))
  > 48:                 valDeltaList = append(valDeltaList, valPair{deltaAmt, valAddr})
  > 49:         }
    50:         // now we sort that list

Reply:
There are actually a few of these that need fixing

Reply:
I recall seeing only two of them during the review, but there might have been more